File: src/main/java/com/health/myapplication/view_pager/ContentsPagerAdapter.java
package com.health.myapplication.view_pager;


import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import com.health.myapplication.fragment.BodyWeight_ChartFragment;
import com.health.myapplication.fragment.BodyWeight_DataFragment;


public class ContentsPagerAdapter extends FragmentStatePagerAdapter {

    private int mPageCount;

    public ContentsPagerAdapter(FragmentManager fm, int pageCount) {
        super(fm);
        this.mPageCount = pageCount;

    }



    @Override

    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                BodyWeight_ChartFragment bodyWeightChartFragment = new BodyWeight_ChartFragment();
                notifyDataSetChanged();
                return bodyWeightChartFragment;

            case 1:
                BodyWeight_DataFragment bodyWeightDataFragment = new BodyWeight_DataFragment();
                notifyDataSetChanged();
                return bodyWeightDataFragment;
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return mPageCount;
    }


    @Override
    public int getItemPosition(Object object) {
        if (object instanceof BodyWeight_ChartFragment) {
            // Create a new method notifyUpdate() in your fragment
            // it will get call when you invoke
            // notifyDatasetChaged();
            ((BodyWeight_ChartFragment) object).changeDataSet();
        } else if((object instanceof BodyWeight_DataFragment)){
            ((BodyWeight_DataFragment) object).update();
        }
        //don't return POSITION_NONE, avoid fragment recreation.
        return super.getItemPosition(object);
    }

    /*
    @Override
    public int getItemPosition(Object object) {
        return POSITION_NONE;
    }
/*
    @Override
    public int getItemPosition(@NonNull Object object) {
        Fragment fragment = (Fragment)object;
        object.
        if(object.)
        return POSITION_NONE;
    }
*/
}

File: src/main/java/com/health/myapplication/view_pager/ContentsPagerAdapter.kt
package com.health.myapplication.view_pager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.health.myapplication.fragment.*

class ContentsPagerAdapter(fa: FragmentActivity, private val mPageCount: Int, private val ACTIVITY_NUMBER: Int) : FragmentStateAdapter(fa) {
    private var ID = -1

    constructor(fa: FragmentActivity, mPageCount: Int, LEVEL: Int, ACTIVITY_NUMBER: Int) : this(fa, mPageCount, ACTIVITY_NUMBER) {
        this.ID = LEVEL
    }

    override fun createFragment(position: Int): Fragment {
        if(ACTIVITY_NUMBER== BODY_WEIGHT_ACTIVITY){
            return when (position) {
                0 -> BodyWeightChartFragment()
                else -> BodyWeightDataFragment()
            }
        }else if(ACTIVITY_NUMBER== CUSTOM_PROGRAM_ACTIVITY){
            return ProgramFragment.newInstance(position + 1, ID)
        }else if(ACTIVITY_NUMBER== RECORD_ACTIVITY){
            return when(position){
                0 -> RecordTodayFragment()
                else -> RecordCalendarFragment()
            }
        }else if(ACTIVITY_NUMBER== GUIDE_ACTIVITY){
            return when(position){
                0 -> ExerciseGuideFragment()
                else -> ExerciseGuideCommunityFragment()
            }
        }else{//PROGRAM_RECOMMEND_ACTIVITY
            return RecommendFragment.newInstance(position + 1, ID)
        }
    }

    override fun getItemCount(): Int {
        return mPageCount
    }

    companion object{
        const val BODY_WEIGHT_ACTIVITY = 600
        const val CUSTOM_PROGRAM_ACTIVITY = 300
        const val PROGRAM_RECOMMEND_ACTIVITY = 200
        const val RECORD_ACTIVITY = 400
        const val GUIDE_ACTIVITY = 100
    }
}